unix/Makefile

```diff
@@ -153,3 +158,9 @@ coverage_test: coverage
 	cd ../tests && MICROPY_MICROPYTHON=../$(DIRNAME)/micropython_coverage ./run-tests --emit native
 	gcov -o build-coverage/py ../py/*.c
 	gcov -o build-coverage/extmod ../extmod/*.c
+
+_frozen_upip.c: frozen_upip/upip.py
+	../tools/make-frozen.py frozen_upip > $@
+
+frozen_upip/upip.py:
+	../tools/bootstrap_upip.sh frozen_upip
```

Comment:
Sure, I noted myself too that it will complicate my work on a beach/plane. I'm happy to duplicate bytes and drop upip tarball (easiest and "most contained" solution) or upip sources into codebase - if you're happy with such duplication.